backend/app/main.py

```diff
@@ -32,4 +41,41 @@ def custom_generate_unique_id(route: APIRoute) -> str:
         allow_headers=["*"],
     )
 
+
+def get_current_user():
+    return {"is_admin": True}  # Mock admin user
+
+
+def get_session():
+    with Session(engine) as session:
+        yield session
+
+
+@app.put("/api/inquiries/{inquiry_id}", response_model=InquiryPublic)
+def edit_inquiry(
+    inquiry_id: UUID,
+    inquiry_update: InquiryCreate,
+    current_user: dict = Depends(get_current_user),
+    session: Session = Depends(get_session),
+):
+    if not current_user["is_admin"]:
+        raise HTTPException(status_code=403, detail="Unauthorized")
+
+    inquiry = session.get(Inquiry, inquiry_id)
+    if not inquiry:
+        raise HTTPException(status_code=404, detail="Inquiry not found")
+
+    if inquiry.responses:
+        raise HTTPException(
+            status_code=400, detail="Cannot edit inquiry with responses"
+        )
+
+    inquiry.text = inquiry_update.text
+    session.add(inquiry)
```

Comment:
Same as above, we talk to the DB through the inquiry service, not in the path operation.

Reply:
addressed